### ISSUE TITLE
Upgrade to Vite 4.2 and add `sourcemapIgnoreList` option

### ISSUE THREAD

**Issue by benmccann** [sveltejs/kit#9306]
### Describe the problem

See https://github.com/nuxt/nuxt/pull/19243

### Describe the proposed solution

Implement in the same manner as https://github.com/nuxt/nuxt/pull/19243. Note that they had a bug with `node_modules` on Windows in that PR, but no support for `node_modules` will be needed if https://github.com/vitejs/vite/pull/12283 makes 4.2

### Importance

nice to have

**Comment by dummdidumm** [comment]
I don't quite understand the description. Reading the Vite PR it sounds like there's nothing to do on our side once that lands in Vite?

**Comment by bluwy** [comment]
I think the idea to hide some of SvelteKit's internal generated code (e.g. those in `.svelte-kit`) from showing up in sourcemap trace by default. That way when an error happens, the user can quickly find where in their own code where the error is happening.

**Comment by benmccann** [comment]
Yep. On our side we will need to set the new option to point to that directory
